File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    answer = ''
    
    i = 1
    while True:
        if food[i] > 1:
            answer += str(i)
            food[i] -= 2
        else:
            if i+1<len(food):
                i += 1
            else:
                break
    answer += '0'
    for i in range(len(answer)-2,-1,-1):
        answer += answer[i]
    return answer

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        String answer = "";
        
        
        for(int i = 1; i < food.length; i++){
            food[i] /= 2;;
        }
        
        for(int i = 1; i < food.length; i++){
            for(int j = 0; j < food[i]; j++){
                answer += i;
            }
        }
        answer += "0";
        for(int i = food.length-1; i > 0; i--){
            for(int j = 0; j < food[i]; j++){
                answer += i;
            }
        }
            
        return answer;
    }
}